sdks/python/apache_beam/pipeline.py

```diff
@@ -396,28 +400,57 @@ def replace_all(self, replacements):
     for override in replacements:
       self._check_replacement(override)
 
-  def run(self, test_runner_api=True):
-    """Runs the pipeline. Returns whatever our runner returns after running."""
-
+  def run(self, test_runner_api=True, runner=None, options=None,
+          interactive=None):
+    """Runs the pipeline. Returns whatever our runner returns after running.
+
+    If another runner instance and options are provided, that runner will
+    execute the pipeline with the given options. If either of them is not set,
+    a ValueError is raised. The usage is similar to directly invoking
+    `runner.run_pipeline(pipeline, options)`.
+    Additionally, an interactive field can be set to override the pipeline's
+    self.interactive field to mark current pipeline as being initiated from an
+    interactive environment.
+    """
+    if interactive:
+      self.interactive = interactive
+    elif (type(self.runner).__module__
+          == 'apache_beam.runners.interactive.interactive_runner' and
+          type(self.runner).__name__ == 'InteractiveRunner'):
```

Comment:
This is the difference from previous [PR](https://www.google.com/url?q=https://github.com/apache/beam/pull/9854). 
All runners are using  "new-style" classes in Python, the `type(obj).__module__/__name__` should always work. Please let me know if there would be backward incompatible cases. 
Thanks!